### PR TITLE
feat: add funding rate to market data

### LINKED ISSUE #9076: [Perps]: Add API for ongoing / current funding rate
Add the missing API for the F

- [ ] add the ongoing / current funding rate as part of the market data event-  this is used to calculate settlement

### PR DESCRIPTION
**Pull request by guoguojin**
Closes #9076

### CODE REVIEW

**Review by candida-d — DISMISSED** [review]

File: core/products/perpetual.go [hunk 1]
```diff
@@ -365,6 +365,15 @@ func (p *Perpetual) handleSettlementCue(ctx context.Context, t int64) {
 	p.startNewFundingPeriod(ctx, t)
 }
 
+func (p *Perpetual) GetFundingRate(t int64) *num.Decimal {
+	internalTWAP := twap(p.internal, p.startedAt, t)
```
Comment (wwestgarth) [inline core/products/perpetual.go:369]: Can you do a `if !p.haveData(t){}` return `0` at the start of this function?

File: datanode/sqlstore/migrations/0025_market_data_add_funding_rate.sql [hunk 2]
```diff
@@ -0,0 +1,83 @@
+-- +goose Up
+
+alter table market_data add column if not exists funding_rate numeric;
+alter table current_market_data add column if not exists funding_rate numeric;
```
Comment (jeremyletang) [inline datanode/sqlstore/migrations/0025_market_data_add_funding_rate.sql:4]: Can / should we make them both nullable, and default to NULL? would make it easily compatible for all existing future market.

File: datanode/gateway/graphql/schema.graphql [hunk 3]
```diff
@@ -256,6 +256,8 @@ type MarketData {
   marketGrowth: String!
   "The last traded price (an unsigned integer)"
   lastTradedPrice: String!
+  "The current funding rate. This applies only to a perpetual market"
+  fundingRate: String!
```
Comment (mattrussell36) [inline datanode/gateway/graphql/schema.graphql:260]: Firstly, what will the value be if this market is not a perp. I would expect a nullable field here?

Secondly, will there be another pr to have this field added to ObservableMarketData too? This is key for frontend requirements

File: datanode/sqlstore/migrations/0025_market_data_add_funding_rate.sql (hunk 2 above)
Comment (mattrussell36) [reply datanode/sqlstore/migrations/0025_market_data_add_funding_rate.sql:4]: I mentioned this in my comment also.. I think returning null over the api would be better than an empty string which is what I would expect given the current schema change.

Comment (guoguojin) [reply datanode/sqlstore/migrations/0025_market_data_add_funding_rate.sql:4]: The nullable and default null value here is already implicit in the declaration. The table is too big to set a not null default, the alter command would take forever to complete on archive nodes.

**Review by jeremyletang — DISMISSED** [review]

**Review by wwestgarth — DISMISSED** [review]

**Review by bglownia — DISMISSED** [review]

File: datanode/gateway/graphql/schema.graphql [hunk 4]
```diff
@@ -320,6 +322,8 @@ type ObservableMarketData {
   marketGrowth: String!
   "The last traded price (an unsigned integer)"
   lastTradedPrice: String!
+  "The current funding rate. This applies only to a perpetual market"
+  fundingRate: String
```
Comment (edd) [inline datanode/gateway/graphql/schema.graphql:326]: **chore (non-blocking)** In the future, let's tidy this object up so that fields that don't apply to all markets don't apply to all markets. I've marked this as non blocking as there'll almost certainly be more than just this field. I'll file a ticket.

**Review by ValentinTrinque — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]
